tests/unit/payment-methods/test-class-upe-payment-gateway.php

```diff
@@ -31,10 +31,25 @@
 use WP_User;
 use Exception;
 
+/**
+ * Overriding global function within namespace for testing
+ */
+function get_woocommerce_currency() {
+	return UPE_Payment_Gateway_Test::$mock_site_currency ? UPE_Payment_Gateway_Test::$mock_site_currency : \get_woocommerce_currency();
+}
```

Comment:
Since I used the global function `get_woocommerce_currency` inside the payment method, I had to do another dirty thing over and override the global function inside the namespace in order to mock out the currency value for tests. Another solution would be to wrap `get_woocommerce_currency` inside a public function inside the `UPE_Payment_Method` class and then I could just mock out the function on the mocked methods. Let me know if you think that's a little less blasphemous and I'll willingly make the change.